### PR TITLE
New module Bio.Data.PDBData to replace/enhance Bio.Data.SCOPData

### PR DESCRIPTION
**Pull request by JoaoRodrigues**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Related to #3981, this PR deprecates `Bio.Data.SCOPData` and centralizes all structural biology conversion tables of 3-letter codes to 1-letter codes in one place. Conversion tables were derived from the PDB component dictionary file, and now include nucleic acids as well.

This PR also deprecates the `three_to_one` and `one_to_three` functions in `Bio.PDB.Polypeptide`.

### CODE REVIEW

**Comment by JoaoRodrigues** [comment]
If @lennax could also have a look at this in her spare time, it would be great, to see if I missed something obvious about the tables or their applications.

**Comment by JoaoRodrigues** [comment]
Had to change the expected result. Turns out `PCA` (pyroglutamic acid) is actually a glutamine derivative, not a glutamic acid derivative. Who would have thunk... (https://en.wikipedia.org/wiki/Pyroglutamic_acid)

**Review by mdehoon — APPROVED** [review]

File: Bio/PDB/Polypeptide.py [hunk 1]
```diff
@@ -194,14 +186,45 @@ def is_aa(residue, standard=False):
...
-        residue = residue.get_resname()
+        residue = f"{residue.get_resname():<3s}"
+    residue = residue.upper()
+    if standard:
+        return residue in protein_letters_3to1
+    else:
+        return residue in protein_letters_3to1_extended
+
+
+def is_na(residue, standard=False):
```
Comment (peterjc) [inline Bio/PDB/Polypeptide.py:198]: Is there a strong reason for this function name? Is seems confusing given the numerical meaning of NA.

Would something like ``is_nuc`` or ``is_nucleotide`` be just as good?

Comment (JoaoRodrigues) [reply Bio/PDB/Polypeptide.py:198]: No, `is_nucleotide` is indeed better!

**Review by peterjc — APPROVED** [review]

**Comment by JoaoRodrigues** [comment]
Thanks @mdehoon & @peterjc !

**Comment by rob-miller** [comment]
Is the message erroneous? 

    'three_to_one' will be deprecated in a future release of Biopython in favor of 'Bio.PDB.Polypeptide.protein_letters_3to1'

Seems like it should be the Dict in Bio/Data/PDBData.py (not Bio/PDB/Polypeptide.py) ?

**Comment by JoaoRodrigues** [comment]
Hi @rob-miller, I linked to the instance of the dictionary in the module, for simplicity's sake. But yeah, `Bio.PDB.Polypeptide.protein_letters_3to1` maps to `Bio/Data/PDBData.py` internally.

**Comment by rob-miller** [comment]
Sorry, must have made a typo on my original import attempt.

Seems preferable for it to live in `Polypeptide` long term, so that is where it should be referenced?  Not really specific to PDB.

**Pull request merged**